permissions.mm

```diff
@@ -367,11 +367,23 @@ void AskForFullDiskAccess(const Napi::CallbackInfo &info) {
 
 // Request Screen Capture Access.
 void AskForScreenCaptureAccess(const Napi::CallbackInfo &info) {
-  if (@available(macOS 10.14, *)) {
-    NSWorkspace *workspace = [[NSWorkspace alloc] init];
-    NSString *pref_string = @"x-apple.systempreferences:com.apple.preference."
-                            @"security?Privacy_ScreenCapture";
-    [workspace openURL:[NSURL URLWithString:pref_string]];
+  if (@available(macOS 10.15, *)) {
+    // Tries to create a capture stream.  This is necessary to add the app back
+    // to the list in sysprefs if the user perviously denied.
+    // https://stackoverflow.com/questions/56597221/detecting-screen-recording-settings-on-macos-catalina
```

Comment:
@glawson in the StackOverflow question, the author says:

> The only reliable way is through `CGDisplayStreamCreate` which is risky as Apple always changes privacy settings every year.

I'm in agreement that this tradeoff is acceptable given the other solutions on the thread seem very brittle and stateful, but I leave that decision up to @codebytere since they are the owner of this repo.